var/spack/repos/builtin/packages/adios2/package.py

```diff
@@ -7,57 +7,58 @@
 
 
 class Adios2(CMakePackage):
-    """Next generation of ADIOS developed in the Exascale Computing Program"""
+    """The Adaptable Input Output System version 2,
+    developed in the Exascale Computing Program"""
 
-    homepage = "https://www.olcf.ornl.gov/center-projects/adios/"
-    url      = "https://github.com/ornladios/ADIOS2/archive/v2.0.0.tar.gz"
-    git      = "https://github.com/ornladios/ADIOS2.git"
+    homepage = "https://csmd.ornl.gov/software/adios2"
+    url = "https://github.com/ornladios/ADIOS2/archive/v2.5.0.tar.gz"
+    git = "https://github.com/ornladios/ADIOS2.git"
 
-    maintainers = ['ax3l', 'chuckatkins']
+    maintainers = ['ax3l', 'chuckatkins', 'williamfgc']
 
     version('develop', branch='master')
     version('2.5.0', sha256='7c8ff3bf5441dd662806df9650c56a669359cb0185ea232ecb3578de7b065329')
     version('2.4.0', sha256='50ecea04b1e41c88835b4b3fd4e7bf0a0a2a3129855c9cc4ba6cf6a1575106e2')
     version('2.3.1', sha256='3bf81ccc20a7f2715935349336a76ba4c8402355e1dc3848fcd6f4c3c5931893')
-    version('2.2.0', sha256='77058ea2ff7224dc02ea519733de42d89112cf21ffe7474fb2fa3c5696152948')
-    version('2.1.0', sha256='d4df3b3035b4236b51c77b59d68e756e825834b2ea3cb17439927a027831453b')
-    version('2.0.0', sha256='4eeedf4404824d8de6e4ef580b8a703c0aedb5c03f900f5ce6f85f0b35980135')
 
+    # general build options
+    variant('mpi', default=True, description='Enable MPI')
+    variant('build_type', default='Release',
+            description='CMake build type',
+            values=('Debug', 'Release', 'RelWithDebInfo', 'MinSizeRel'))
     variant('shared', default=True,
             description='Also build shared libraries')
     variant('pic', default=True,
             description='Enable position independent code '
                         '(for usage of static in shared downstream deps)')
-    variant('mpi', default=True,
-            description='Enable MPI')
+    variant('endian_reverse', default=False,
```

Comment:
I am not sure if we can do that here or in a `packages.yaml`, but changing the default depending on `arch` to `default=True` for big endian machines would be cool. :)

Reply:
That confuses me. Shouldn't the `.bp` file format detect such moved data sets?
I had this situation in the past quite naturally at blue gene q + support systems, feeding of the same filesystem.

Reply:
@ax3l yes, the bp format always encodes the endianness, this option is for the reader (not the writer) to reverse the endianness if necessary. Keeping the default as OFF is about the probability of mixing endianness + lack of testing from our side.

Reply:
Migrated off-topic discussion to https://github.com/ornladios/ADIOS2/issues/1822